static/src/js/util/accessMethods/buildAccessMethods.js

```diff
@@ -0,0 +1,196 @@
+import { camelCase, uniq } from 'lodash'
+
+import { isDownloadable } from '../../../../../sharedUtils/isDownloadable'
+import { generateFormDigest } from './generateFormDigest'
+import { getVariables } from './getVariables'
+import { supportsBoundingBoxSubsetting } from './supportsBoundingBoxSubsetting'
+import { supportsShapefileSubsetting } from './supportsShapefileSubsetting'
+import { supportsTemporalSubsetting } from './supportsTemporalSubsetting'
+import { supportsVariableSubsetting } from './supportsVariableSubsetting'
+
+/**
+ * Builds the different access methods available for the provided collection
+ * @param {object} collectionMetadata Collection Metadata
+ * @param {boolean} isOpenSearch Is the collection an open search collection
+ * @returns {object} Access methods
+ */
+export const buildAccessMethods = (collectionMetadata, isOpenSearch) => {
+  const {
+    granules = {},
+    services = {},
+    variables: associatedVariables = {}
+  } = collectionMetadata
+
+  const accessMethods = {}
+
+  let harmonyIndex = 0
+
+  const { items: serviceItems = null } = services
+
+  if (serviceItems !== null) {
+    serviceItems.forEach((serviceItem) => {
```

Comment:
Could this `forEach` be split apart into smaller chunks? The first section could be simplified with a `filter`, and the each of the `if` feel like they might be able to be broken out into smaller functions

Reply:
Yes I think pulling the `if` statement's code out into smaller utils would be fine, I'm just not sure if I have time to do it before this has to be merged.